src/Client/RepoFileFetcherInterface.php

```diff
@@ -16,14 +16,16 @@ interface RepoFileFetcherInterface
      *   The name of the metadata file to fetch.
      * @param integer $maxBytes
      *   The maximum number of bytes to download.
+     * @param array $options
+     *   (optional) Additional fetcher-specific options.
      *
      * @return \GuzzleHttp\Promise\PromiseInterface
      *   A promise representing the eventual result of the operation. If
      *   successful, the promise should wrap around an instance of
      *   \Psr\Http\Message\StreamInterface, which provides a stream of the
      *   retrieved data.
      */
-    public function fetchMetaData(string $fileName, int $maxBytes): PromiseInterface;
+    public function fetchMetaData(string $fileName, int $maxBytes, array $options = []): PromiseInterface;
```

Comment:
@phenaproxima what do you think about not adding `$options` to the interface methods but just GuzzleFileFetcher since this is an optional argument it would work. because other fetchers might want to pass there args in a different manner. Like if a different backend had an options class. or even if they just wanted to add a couple of boolean flag arguments

Reply:
Sure, that works! Nice idea.